apps/token-e2e/src/support/staking.functions.js

```diff
@@ -48,32 +49,36 @@ Cypress.Commands.add('staking_validator_page_remove_stake', (stake) => {
     .click();
 });
 
-Cypress.Commands.add(
-  'staking_page_associate_tokens',
-  (amount, approve = false) => {
-    cy.highlight(`Associating ${amount} tokens`);
-    cy.get(ethWalletAssociateButton).first().click();
+Cypress.Commands.add('staking_page_associate_tokens', (amount, options) => {
+  let approve = options && options.approve ? options.approve : false;
+  let type = options && options.type ? options.type : 'wallet';
+
+  cy.highlight(`Associating ${amount} tokens`);
+  cy.get(ethWalletAssociateButton).first().click();
+  if (type === 'wallet') {
```

Comment:
chore (non0-blocking): can we enum these rather than having a magic string please

Reply:
We could, but enum is TS, and this is JS :D